idea-plugin/build.gradle

```diff
@@ -0,0 +1,59 @@
+apply plugin: 'java'
+apply plugin: 'com.palantir.external-publish-intellij'
+apply plugin: 'org.jetbrains.grammarkit'
+
+intellij {
+    pluginName = "gradle-consistent-versions"
+    updateSinceUntilBuild = true
+    version = "2024.1"
+    plugins = ['java', 'org.jetbrains.plugins.gradle']
+}
+
+patchPluginXml {
+    pluginDescription = "Language plugin for version.props allowing for auto-lock updating"
+    sinceBuild = '241' // TODO: test against this version of IntelliJ to ensure no regressions
+    untilBuild = ''
+}
+
+// Javadoc fails if there are no public classes to javadoc, so make it stop complaining.
+tasks.named("javadoc", Javadoc) {
+    failOnError = false
+}
+
+sourceSets.main.java.srcDirs 'src/main/generated'
+
+javaVersion {
+    target = 17
+    runtime = 17
+}
+
+dependencies {
+    implementation 'com.fasterxml.jackson.dataformat:jackson-dataformat-xml'
+    annotationProcessor 'org.immutables:value'
+    compileOnly 'org.immutables:value::annotations'
+    testImplementation 'org.junit.jupiter:junit-jupiter'
+}
+
+grammarKit {
+    jflexRelease = "1.9.2"
+    grammarKitRelease = "2022.3.2"
+}
+
+generateParser {
+    sourceFile = file('src/main/java/com/palantir/gradle/versions/intellij/VersionProps.bnf')
+    targetRootOutputDir = file('src/main/generated')
```

Comment:
I think we should put the generate files in the build directory. The convention is in `build/generated/sources/intellij-parser/main/java`

Reply:
This means you can remove the line from the .gitignore too.

Reply:
Have moved - I still seem to need `sourceSets.main.java.srcDirs 'build/generated/sources/intellij-parser/main/java'` is there a way to remove this?